src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Extensions/DefaultTagHelperTargetExtension.cs

```diff
@@ -650,10 +646,13 @@ private string GetContent(HtmlContentIntermediateNode node)
     // Internal for testing
     internal static string GetDeterministicId(CodeRenderingContext context)
     {
-        // Use the file checksum along with the absolute position in the generated code to create a unique id for each tag helper call site.
-        var checksum = ChecksumUtilities.BytesToString(context.SourceDocument.Text.GetChecksum());
-        var uniqueId = checksum + context.CodeWriter.Location.AbsoluteIndex;
-
+        var uniqueId = (string)context.Items[CodeRenderingContext.SuppressUniqueIds];
```

Comment:
This is going to conflict with https://github.com/dotnet/razor/pull/10764 when it goes in.